R/annotate_cn_matrix_by_genes.R

```diff
@@ -0,0 +1,53 @@
+#' @title Annotate copy number matrix by gene names
+#' 
+#' @description Annotates a copy number matrix using user-provided gene names 
+#' 
+#' @param gene_ids The "gene_id" column stores gene symbols (characters).
+#' @param cn_matrix User can provide a matrix of CN values for the samples in the metadata.
+#' See [GAMBLR.utils::segmented_data_to_cn_matrix] for more information on how to create this matrix.
+#' 
+#' @return A copy number matrix with sample ids as rows and gene names as columns
+#' 
+#' @import dplyr 
+#' @export
+#'
+#' @examples
+#' 
+#' \dontrun{
+#' # Get sample metadata 
+#' dlbcl_meta = get_gambl_metadata() %>% filter(pathology == "DLBCL")
+#' 
+#' # Consider some genes
+#' gene_symbols = c("MYC", "MIR17HG", "CCND3","ID3","DDX3X", "SYNCRIP")
+#' 
+#' # Get cn_matrix
+#' all_segments = get_cn_segments(these_samples_metadata = dlbcl_meta)
+#' cn_matrix = segmented_data_to_cn_matrix(
+#'    seg_data = all_segments,
+#'     strategy="auto_split",
+#'    these_samples_metadata = dlbcl_meta,
+#'    adjust_for_ploidy=TRUE
+#' )
+#' 
+#' annotate_cn_matrix_by_genes (gene_ids = gene_symbols,
+#'                              cn_matrix = cn_matrix
```

Comment:
Is this functionality meant to replace what already happens within the pretty_CN_heatmap function? If so, will that code be replaced by a call to this function?